tools/codegen/dest/register_dispatch_key.py

```diff
@@ -85,6 +92,51 @@ def __call__(self, f: Union[NativeFunctionsGroup, NativeFunction]) -> List[str]:
         else:
             assert_never(f)
 
+    def wrapper_kernel_sig(self, f: NativeFunction) -> Union[NativeSignature, DispatcherSignature]:
+        # The prefix is just to ensure uniqueness. The Dispatcher API doesn't guarantee unique kernel names.
+        return kernel_signature(f, self.backend_index, prefix=f'wrapper_{f.func.name.overload_name}_')
```

Comment:
Just a high level observation and not to overengineer, but here `prefix` is derived from `f` in a pretty straightforward way. If that's true everywhere, then it might be cleaner to host the prefix-making logic in `NativeFunction`.

Reply:
So, let me know if we disagree, but the value of `prefix` here seems more to me like an implementation detail of how we code-generate the stuff that we register to the dispatcher than a property of a NativeFunction. Rather than directly registering native kernels to the dispatcher, the codegen creates anonymous wrappers that call into to the native kernels, and registers those wrappers instead. We just append `wrapper_` to each wrapper function's name for clarity, and to avoid name collisions with the namespaced definitions that we generate in the same file.

So I kind of see `NativeSignature`/`DispatcherSignature` as a lens into how to display a given NativeFunction's signature, like what schema and naming transformations to apply, and the wrapper prefix is just another knob along those same lines.

Reply:
Ah sorry! Not `NativeFunction` but rather `NativeSignature` and `DispatchSignature`. You're right that prefixes shouldn't be part of `NativeFunction`'s world. But the only "free" value here is the 'wrapper_{}_' - the function name in the middle comes from `f`, which is 1-1 with whichever Signature `wrapper_kernel_sig()` constructs - so making the caller supply the correspondence here would only be useful if we ever wanted to break that relationship, which I don't think we would, right? It seems more like an opportunity for the caller to mess things up by passing a constant string or whatever.

Anyway to cut to the chase, the whole correct by construction thing would say, only pass the constant part of the prefix here and onward into the Signatures, and make them (the Signatures) responsible for folding `f.func.name.overload_name` into nonempty prefixes. 

(BTW if you don't actually use multiple prefixes, you could just have the Signatures carry a bit saying whether to mangle in the overload name. (And if `kernel_signature()` is always called with a nonempty prefix, you could just have it unilaterally construct the Signature instances with prefixes turned on.)